### PR TITLE
session_test: move tests

### PR DESCRIPTION
**Pull request by hanfei1991**
This test will retry forever in mock-tikv. It seems to exist a lock that would never be resolved. I need a help @coocood @tiancaiamao

### CODE REVIEW

**Comment by shenli** [comment]
@hanfei1991 Please fix CI.

**Comment by hanfei1991** [comment]
@shenli Please look at the description

**Comment by shenli** [comment]
Got it!

File: new_session_test.go [hunk 1]
```diff
@@ -558,6 +558,99 @@ func (s *testSessionSuite) TestAutoIncrementID(c *C) {
...
+	tk.MustQuery("select c1 from t where c2 = 11").Check(testkit.Rows("6"))
+	currLastInsertID := tk.Se.GetSessionVars().PrevLastInsertID
+	c.Assert(lastInsertID+5, Equals, currLastInsertID)
+
+	// insert set
+	lastInsertID = currLastInsertID
+	tk.MustExec("begin")
+	tk.MustExec("insert into t set c2 = 31")
+	tk.MustQuery("select c1 from t where c2 = 31").Check(testkit.Rows("9"))
+	tk.MustQuery("update t set c2 = 44 where c2 = 2")
```
Comment (tiancaiamao) [inline new_session_test.go]: MustExec

File: new_session_test.go [hunk 2]
```diff
@@ -558,6 +558,99 @@ func (s *testSessionSuite) TestAutoIncrementID(c *C) {
...
+	tk.MustExec("begin")
+	tk.MustExec("prepare stmt from 'insert into t (c2) values (?)'")
+	tk.MustExec("set @v1=100")
+	tk.MustExec("set @v2=200")
+	tk.MustExec("set @v3=300")
+	tk.MustExec("execute stmt using @v1")
+	tk.MustExec("execute stmt using @v2")
+	tk.MustExec("execute stmt using @v3")
+	tk.MustExec("deallocate prepare stmt")
+	tk.MustQuery("select c1 from t where c2 = 12").Check(testkit.Rows())
```
Comment (tiancaiamao) [inline new_session_test.go]: `testkit.Rows("7")`

File: new_session_test.go [hunk 3]
```diff
@@ -558,6 +558,99 @@ func (s *testSessionSuite) TestAutoIncrementID(c *C) {
...
+	tk.MustExec("update t set c2 = 111 where c2 = 5")
+
+	tk1.MustExec("update t set c2 = 222 where c2 = 5")
+
+	tk.MustExec("commit")
+
+	tk.MustQuery("select c1 from t where c2 = 12").Check(testkit.Rows())
```
Comment (tiancaiamao) [inline new_session_test.go]: `testkit.Rows("7")`

**Comment by hanfei1991** [comment]
@tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**